### PR TITLE
[libc][stdlib] Add calloc header

### PR DESCRIPTION
**Pull request by PiJoules**

### CODE REVIEW

File: libc/src/stdlib/calloc.h [hunk 1]
```diff
@@ -0,0 +1,20 @@
+//===-- Implementation header for calloc ------------------------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include <stdlib.h>
```
Comment (michaelrj-google) [inline libc/src/stdlib/calloc.h:9]: this can be `stddef` instead of `stdlib`

**Review by michaelrj-google — APPROVED** [review]
LGTM with one nit

File: libc/src/stdlib/calloc.h (hunk 1 above)
Comment (PiJoules) [reply libc/src/stdlib/calloc.h:9]: Done.

**Comment by michaelrj-google** [comment]
if you're adding this it might be worthwhile to also add a header for `realloc`

**Pull request merged**